reports-backend/src/main/java/org/jboss/da/reports/impl/ReportsGeneratorImpl.java

```diff
@@ -116,4 +126,57 @@ private void addDependencyReports(ArtifactReport ar, Set<GAVDependencyTree> depe
         }
     }
 
+    private AdvancedArtifactReport generateAdvancedArtifactReport(ArtifactReport report) {
+        AdvancedArtifactReport advancedReport = new AdvancedArtifactReport();
+        Set<GAV> modulesAnalyzed = new HashSet<>();
+        populateAdvancedArtifactReportFields(advancedReport, report, modulesAnalyzed);
+        return advancedReport;
+    }
+
+    private void populateAdvancedArtifactReportFields(AdvancedArtifactReport advancedReport,
+            ArtifactReport report, Set<GAV> modulesAnalyzed) {
+        for (ArtifactReport dep : report.getDependencies()) {
+            if (modulesAnalyzed.contains(dep.getGav())) {
+                // if module already analyzed, skip
+                continue;
+            } else if (isDependencyAModule(report, dep)) {
+                // if dependency is a module, but not yet analyzed
+                modulesAnalyzed.add(dep.getGav());
+                populateAdvancedArtifactReportFields(advancedReport, dep, modulesAnalyzed);
+            } else {
+                // only generate populate advanced report with community GAVs
+                if (VersionParser.isRedhatVersion(dep.getVersion()))
+                    continue;
+
+                // we have a top-level module dependency
+                if (dep.isWhitelisted())
+                    advancedReport.addWhitelistArtifact(dep.getGav());
+                if (dep.isBlacklisted())
+                    advancedReport.addBlacklistArtifact(dep.getGav());
+
+                if (dep.getBestMatchVersion().isPresent()) {
+                    advancedReport.addCommunityGavWithBestMatchVersion(dep.getGav());
+                } else if (!dep.getAvailableVersions().isEmpty()
+                        || hasWhitelistedGA(dep.getGroupId(), dep.getArtifactId())) {
+                    advancedReport.addCommunityGavWithBuiltVersion(dep.getGav());
+                } else {
+                    advancedReport.addCommunityGav(dep.getGav());
+                }
+            }
+        }
+    }
+
+    private boolean hasWhitelistedGA(String groupId, String artifactId) {
+        if (whiteArtifactService.getAll() == null) {
+            return false;
+        }
+        return whiteArtifactService.getAll().stream()
+                .anyMatch(white -> white.getArtifactId().equals(artifactId) && white.getGroupId().equals(groupId));
+    }
+
+    private boolean isDependencyAModule(ArtifactReport artifactReport, ArtifactReport dependency) {
```

Comment:
:+1: Do it the way with checking if it is in the same SCM and create and issue to find a better way.